app/replica.go

```diff
@@ -233,19 +234,24 @@ func startReplica(c *cli.Context) error {
 		status := s.Replica().GetCloneStatus()
 		if status != "completed" {
 			logrus.Infof("Set clone status as inProgress")
-			s.Replica().SetCloneStatus("inProgress")
-			if err = CloneReplica(s, "tcp://"+address, cloneIP, snapName); err != nil {
-				logrus.Infof("Set clone status as error")
-				s.Replica().SetCloneStatus("error")
+			if err := s.Replica().SetCloneStatus("inProgress"); err != nil {
 				return err
 			}
+			if err = CloneReplica(s, "tcp://"+address, cloneIP, snapName); err != nil {
+				logrus.Error("Error in cloning replica, found error ", err.Error())
+				return s.Replica().SetCloneStatus("error")
```

Comment:
this overwrites the error that got for CloneReplica() API. Needs a fix here.

Reply:
fixed